### DISCUSSION TITLE
How to Implement RLP

### DISCUSSION THREAD

**Discussion by UrsaMajor-t** [Consensys/gnark#495] [Q&A]
RLP decoding of data in the circuit, is there an example of implementation?

**Comment by gbotrel** [comment]
Didn't see a RLP decoding snippet; some thoughts; if the length & types the input is not known, any decoding-in-circuit will be quite expensive. It might make more sense to do the decoding out-of-circuit and provide decoded inputs as a witness to the circuit.
